src/main/java/org/mskcc/cbio/oncokb/web/rest/CronJobController.java

```diff
@@ -180,6 +199,77 @@ public void updateTokenStats() {
+     * 
+     * @throws IOException
+     */
+    @GetMapping(path = "/user-usage-analysis")
+    public void analyzeUserUsage() throws IOException {
+        log.info("User usage analysis started...");
+        List<UserTokenUsageWithInfo> tokenStats =  tokenStatsService.getTokenUsageAnalysis(Instant.now().minus(365, ChronoUnit.DAYS));    
+        UsageSummary resourceSummary = new UsageSummary();       
+        Map<Long, UserUsage> userSummary = new HashMap<>();
+
+        for (UserTokenUsageWithInfo state: tokenStats){
+            calculateUsageSummary(resourceSummary, state);
+
+            long userId = state.getToken().getUser().getId();
+            User user = state.getToken().getUser();
+            if (!userSummary.containsKey(userId)){
+                userSummary.put(userId, new UserUsage());
+                UserUsage currentUsage = userSummary.get(userId);
+                currentUsage.setUserFirstName(user.getFirstName());
+                currentUsage.setUserLastName(user.getLastName());
+                currentUsage.setUserEmail(user.getEmail());
+                currentUsage.setSummary(new UsageSummary());
+                Optional<UserDetails> userDetails = userDetailsRepository.findOneByUser(user);
+                if (userDetails.isPresent()){
+                    currentUsage.setJobTitle(userDetails.get().getJobTitle());
+                    currentUsage.setCompany(userDetails.get().getCompany());
+                    currentUsage.setLicenseType(userDetails.get().getLicenseType().getName());
+                }
+            }
+            UsageSummary currentUsageSummary = userSummary.get(userId).getSummary();
+            calculateUsageSummary(currentUsageSummary, state);
+        }
+        
+        ObjectMapper mapper = new ObjectMapper();
+        File resourceResult = new File("./resourceSummary.json");
+        File userResult = new File("./userSummary.json");
+        mapper.writeValue(resourceResult, resourceSummary);
+        mapper.writeValue(userResult, userSummary);
+
+        String s3AccessKey = applicationProperties.getAws().getS3AccessKey();
+        String s3SecretKey = applicationProperties.getAws().getS3SecretKey();
+        String s3Region = applicationProperties.getAws().getS3Region();
+        AWSCredentials credentials = new BasicAWSCredentials(s3AccessKey, s3SecretKey);
+        AmazonS3 s3client = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(credentials)).withRegion(s3Region).build();
+        s3client.putObject("oncokb", RESOURCES_USAGE_SUMMARY_FILE, resourceResult);
+        s3client.putObject("oncokb", USERS_USAGE_SUMMARY_FILE, userResult);
+        resourceResult.delete();
+        userResult.delete();
+        log.info("User usage analysis completed!");
+    }
+
+    private void calculateUsageSummary(UsageSummary usageSummary, UserTokenUsageWithInfo info){
+        String resource = info.getResource().split(" ")[1];
```

Comment:
This is maybe out of this pr's scope. But I think we should have a string builder when we deal with resource. So that we have all logics together. Even better if we could model the resource.